src/App/Header/UserMenu/UserMenu.tsx

```diff
@@ -0,0 +1,24 @@
+import { Authenticator, useAuthenticator } from "@aws-amplify/ui-react";
+import { Popover } from "@headlessui/react";
+import Icon from "components/Icon";
+import Menu from "./Menu";
+
+export default function UserMenu() {
+  const { user, signOut } = useAuthenticator((context) => [context.route]);
```

Comment:
Found a bug, repro steps below:
1. Open webapp (Marketplace)
2. Click Login button
3. Create profile if non-existent
4. Once profile is created, sign in
5. Reload page
6. Verify that you're still logged in
7. Click on the profile icon in top right (to sign out)
8. Click "Sign Out" button
Expected: user logged out
Actual: nothing happens
9. Reload page again -OR- manually type in a new URL in the URL bar
10. User is now logged out for real

Repro video below (first part shows regular sign-in/sign-out, the second part shows the bug)
(ignore Loom notifications):
https://www.loom.com/share/71503d8f99e24168aa8a6ae44da89801?sid=ac5ff299-b82e-4d2c-a0e8-66c0eba5cdfb

Reply:
it seems `useAuthenticator.user` is not good indicator if user is logged in.
fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2427/commits/736ebd64e871d995151f5e4c9776a862d68edaca